src/kumo-api.ts

```diff
@@ -483,8 +486,8 @@ export class KumoApi {
       return null;  
     }
     
-    if (!data || data == '{ _api_error: \'device_authentication_error\' }') {
-      this.log.warn('Kumo API: error direct querying device: %s.', serial);
+    if(!data || '_api_error' in data) {
+      this.log.warn('Kumo API: error direct querying device: %s; %s.', serial, data);
```

Comment:
There's one more hole I noticed in operation - it appears that api_error responses aren't getting caught by this line and are falling through to the return and being caught by the caller function with an error:
```
Kumo API: bad response from queryDevice_Direct - { _api_error: 'device_authentication_error' }
```

Once again, really tough to test locally. @fjs21 does this look like the correct way to check if the json response contains an _api_error?